### PR TITLE
Covert latex drawer to use _utils.get_instructions()

### LINKED ISSUE #1273: latex_source visualizer should not use transpiler(format='json')
We are going to remove the option `format='json'` from the transpiler (see #1129). For that, we need to move the visualizers into the function `get_instractions` introduced in PR #1187.

The function `_generate_latex_source` calls `transpile_dag(dag_circuit, basis_gates=basis, format='json')`. This needs to be removed in favor of `_utils.get_instructions(dag)`.

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit converts the latex drawer to use the
_utils.get_instructions() function instead of using the transpiler. This
allows us to remove our dependency on the transpiler for circuit
visualization and also makes our bit ordering consistent with the
documented ordering.

### Details and comments

Fixes #1273

### CODE REVIEW

File: qiskit/tools/visualization/_circuit_visualization.py [hunk 1]
```diff
@@ -521,8 +521,9 @@ def _generate_latex_source(circuit, filename=None,
         str: Latex string appropriate for writing to file.
     """
     dag_circuit = DAGCircuit.fromQuantumCircuit(circuit, expand_gates=False)
-    json_circuit = transpile_dag(dag_circuit, basis_gates=basis, format='json')
-    qcimg = _latex.QCircuitImage(json_circuit, scale, style=style,
```
Comment (1ucian0) [inline qiskit/tools/visualization/_circuit_visualization.py]:
```suggestion
 
    # Deprecate the following lines     -------------------------------------------------|
    if basis is None:                                                                  # |
        basis = ("id,u0,u1,u2,u3,x,y,z,h,s,sdg,t,tdg,rx,ry,rz,"                        # |
                 "cx,cy,cz,ch,crz,cu1,cu3,swap,ccx,cswap")                             # |
    else:                                                                              # |
        warnings.warn('The basis kwarg is deprecated and the circuit drawer '          # |
                      'function will not be able to adjust basis gates itself '        # |
                      'in a future release', DeprecationWarning)                       # |
    basis = basis.split(',') if basis else []                                          # |
    dag = _dagunroller.DagUnroller(dag, _dagbackend.DAGBackend(basis)).expand_gates()  # |
    #     -------------------------------------------------------------------------------|
```

Comment (mtreinish) [reply qiskit/tools/visualization/_circuit_visualization.py]: Well, @ajavadia said that we shouldn't do this here, he said that he had a plan to retain the basis functionality after #1210 and the other transpiler/dag work he's working on for 0.7. If we add it here (and in the other drawer methods) we'll be adding more places for him to cleanup later. So I'm going to defer to him and we can wait for that work to finish up.

Also, the warning will be duplicated here, we already warn once in circuit_drawer() if basis is set.

Comment (1ucian0) [reply qiskit/tools/visualization/_circuit_visualization.py]: I removed the warning here: https://github.com/Qiskit/qiskit-terra/pull/1280/files#diff-601179d490e381950db4bf9a4fe01d93R356

**Comment by jaygambetta** [comment]
@mtreinish what about this one

**Comment by mtreinish** [comment]
I still need to fix this patch, it's a bit more of a rats nest here and it's not quite working yet. The change in the format for the registers is messing things up a bit. I'm going to work on it right now, I put it off and prioritized the easier patches first.

**Comment by mtreinish** [comment]
@jaygambetta I think I've fixed most of the bugs. But I'm still not super confident with this, there are a lot of edge cases hidden in the code (I really want to rewrite the whole thing, but that will have to wait) So if you and other people can really put this through it's paces and if you encounter anything that looks wrong comment here with the circuit code and I'll try to fix those issues.

**Comment by ajavadia** [comment]
I found that conditionals are not drawn correctly with this patch.
This code:
```python
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
q = QuantumRegister(3, 'q')
c = ClassicalRegister(1, 'c')
circ = QuantumCircuit(q, c)
circ.u2(1.57, 0.7, q[0])
circ.measure(q[0], c[0])
circ.u1(0.1, q[1]).c_if(c, 1)
circ.draw(output='latex')
```
is drawn correctly like this on master (except the bit order of course):
image

but with this patch the conditional vertical wire is gone:
image

This is also an issue with the mpl drawer (maybe due to #1278 ?)
image

But the text drawer is fine:
```
q_2: |0>──────────────────────────────
                           ┌─────────┐
q_1: |0>───────────────────┤ U1(0.1) ├
        ┌──────────────┐┌─┐└────┬────┘
q_0: |0>┤ U2(1.57,0.7) ├┤M├─────┼─────
        └──────────────┘└╥┘┌────┴────┐
 c_0: 0 ═════════════════╩═╡   = 1   ╞
                           └─────────┘
```

**Pull request merged**